### ISSUE TITLE
Update config.guess and config.sub files

### ISSUE THREAD

**Issue by ryandesign** [opencog/link-grammar#1492]
Even your latest 5.12.4 release from just a few weeks ago contains config.guess and config.sub files dated 2013. Please ensure, prior to each release, that you use `autoreconf -fvi` or equivalent to get the latest config.guess and config.sub and other generated files into the release.

**Comment by linas** [comment]
Oh, ah huh. I just now did this on my machine.  I fixed this meatball surgery style in #1516 -- if this is wrong, let me know. If you happen to have a modern, recommended version of `autogen.sh`, (and I think you might, cause you do a lot of porting work) please open a pull req with that in it.  The current `autogen.sh` is at least 20 years old, some copy of a copy of a copy lost in the sands of time.

**Comment by ampli** [comment]
Now that `autogen.sh` invokes `autoreconf -fvi`,  the calls to `libtoolize`/`glibtoolize` , `aclocal`, `automake`, and  `autoconf` just repeat its work and are not needed. I can send a PR with a fix for that and other fixes I gathered.


``` text
$ sh -x ./autogen.sh --no-configure                                                                                                                                                                                                                                                              LG (small-fixes *+$|u+6-5)
+ rm -f autogen.err
+ automake --version
+ perl -ne 'if (/\(GNU automake\) (([0-9]+).([0-9]+))/) {print; if ($2 < 1 || ($2 == 1 && $3 < 4)) {exit 1;}}'
automake (GNU automake) 1.16.5
+ '[' 0 -ne 0 ']'
++ aclocal --print-ac-dir
+ test '!' -d /usr/share/aclocal
+ autoreconf -fvi
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I m4
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: aclocal --force -I m4
autoreconf: running: /usr/bin/autoconf --force
autoreconf: configure.ac: not using Autoheader
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:83: installing './compile'
configure.ac:56: installing './missing'
bindings/java-jni/Makefile.am: installing './depcomp'
autoreconf: Leaving directory '.'
+ case `uname` in
++ uname
+ libtoolize --force --copy
libtoolize: putting auxiliary files in '.'.
libtoolize: copying file './ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: copying file 'm4/libtool.m4'
libtoolize: copying file 'm4/ltoptions.m4'
libtoolize: copying file 'm4/ltsugar.m4'
libtoolize: copying file 'm4/ltversion.m4'
libtoolize: copying file 'm4/lt~obsolete.m4'
+ echo 'Creating aclocal.m4: aclocal  '
Creating aclocal.m4: aclocal  
+ aclocal
+ echo 'automake --add-missing --copy --foreign'
automake --add-missing --copy --foreign
+ automake --add-missing --copy --foreign
+ '[' -f config.cache ']'
+ echo 'Creating configure...'
Creating configure...
+ autoconf
+ run_configure=true
+ for arg in $*
+ case $arg in
+ run_configure=false
+ false
+ echo

+ echo 'Now run '\''configure'\'' and '\''make'\'' to compile link-grammar.'
Now run 'configure' and 'make' to compile link-grammar.
```
